plutus-core/cost-model/budgeting-bench/Benchmarks/Bitwise.hs

```diff
@@ -0,0 +1,83 @@
+-- editorconfig-checker-disable-file
+
+{-# LANGUAGE TypeOperators #-}
+
+module Benchmarks.Bitwise (makeBenchmarks) where
+
+import Common
+import Generators
+
+import PlutusCore
+import PlutusCore.Evaluation.Machine.ExMemoryUsage
+
+import Criterion.Main
+import Data.ByteString qualified as BS
+import Hedgehog qualified as H
+
+
+---------------- ByteString builtins ----------------
+
+-- Smallish bytestring inputs: 150 entries.  Note that the length of a
+-- bytestring is eight times the size.
+smallerByteStrings150 :: H.Seed -> [BS.ByteString]
+smallerByteStrings150 seed = makeSizedByteStrings seed [1..150]
+
+-- Make an integer of size n which encodes to 0xFF...FF
+allFF :: Int -> Integer
+allFF n = 256^(8*n) - 1
```

Comment:
> What is 256? Is it not `2^(8*n) - 1`?

No, that's right.  We're looking for _size_ n here, which means 8*n bytes (thanks again to the infuriating fact that we measure sizes in 8-byte words).  Since that's a number of bytes, the total number of _bits_ is 8*8*n, and 2^(8*8*n) = 256^(8*n) is 1000...000 with 64*n zero bits.  Subtracting 1 from that removes the 1 at the front and changes all the remaining bits to 1, so you've got 64*n 1 bits, or 8*n 0xFF bytes, which is what you want.  Thanks for making me check though: it's very easy to get this kind of thing wrong.